src/components/Input/index.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { forwardRef } from "react";
+import useIsDeprecatedForm from "../../hooks/internal/useIsDeprecatedForm";
+import { ComponentWithProperties } from "../../types";
+import { ComponentName } from "../componentName";
+import { default as InputInternal } from "./Input";
+import { default as InputDeprecated } from "./Input_deprecated";
+import {
+  DeprecatedInputProps,
+  InputProps,
+  setDeprecatedProps as shouldUseDeprecatedInput,
+} from "./types";
+
+const { TYPES, VARIANTS, COLORS, DEFAULT_PROPS } = InputDeprecated;
+
+const Input = forwardRef(function InputComponent(props: InputProps, ref) {
+  const isDeprecatedForm = useIsDeprecatedForm();
+
+  if (shouldUseDeprecatedInput(props, isDeprecatedForm)) {
+    return <InputDeprecated {...(props as DeprecatedInputProps)} />;
```

Comment:
Isn't the idea of the `shouldUseDeprecatedInput` type guard is avoid doing "as ..."?
I'd expect to either use the type guard or to do "as ...", but why both?

Reply:
Yep I changed something in between but when I reverted it I forgot to remove this